js/piwik.js

```diff
@@ -2605,10 +2609,34 @@ if (typeof Piwik !== 'object') {
                 callback();
             }
 
+            /**
+             * Send each bulk tracks to the server
+             * @return {void}
+             */
+            function sendBulkTrackingRequests() {
+
+                // Work on a local copy
+                var tmp = JSON.parse(JSON.stringify(bulkTrackingCollection));
+                bulkTrackingCollection.length = 0;
+
+                // Do a request only if we have to
+                if(tmp.length >= 1) {
+                  var bulk = '{"requests":["?' + tmp.join('","?') + '"]}';
+                  sendRequest(bulk, configTrackerPause, null, true);
+                }
+            }
+
             /*
              * Send request
              */
-            function sendRequest(request, delay, callback) {
+            function sendRequest(request, delay, callback, force) {
+
+
+                if(enableBulkTracking && !force) {
```

Comment:
I think there needs to be maybe an additional check for `if !configDoNotTrack`. If `configDoNotTrack=true` it should not track anything.